internal/databases/mongo/binary/mongod.go

```diff
@@ -0,0 +1,214 @@
+	).Decode(&versionHolder)
+
+	return versionHolder.Version, err
+}
+
+type MongodConfig struct {
+	Net struct {
+		BindIP string `bson:"bindIp" json:"bindIp"`
+		Port   int    `bson:"port" json:"port"`
+	} `bson:"net" json:"net"`
+	Storage struct {
+		DBPath string `bson:"dbPath" json:"dbPath"`
+	} `bson:"storage" json:"storage"`
+}
+
+func (mongodService *MongodService) MongodConfig() (*MongodConfig, error) {
+	getCmdLineOpts := struct {
+		Parsed MongodConfig `bson:"parsed" json:"parsed"`
+	}{}
+	err := mongodService.MongoClient.Database(adminDB).RunCommand(
+		mongodService.Context,
+		bson.M{"getCmdLineOpts": 1},
+	).Decode(&getCmdLineOpts)
+
+	return &getCmdLineOpts.Parsed, err
+}
+
+func (mongodService *MongodService) GetReplSetName() (string, error) {
+	replSetNameHolder := struct {
+		ReplSetName string `bson:"setName"`
+	}{}
+	err := mongodService.MongoClient.Database(adminDB).RunCommand(
+		mongodService.Context,
+		bson.M{"isMaster": 1},
+	).Decode(&replSetNameHolder)
+
+	return replSetNameHolder.ReplSetName, err
+}
+
+func (mongodService *MongodService) GetBackupCursor() (*mongo.Cursor, error) {
+	return mongodService.MongoClient.Database(adminDB).Aggregate(mongodService.Context, mongo.Pipeline{
+		{{Key: "$backupCursor", Value: bson.D{}}},
+	})
+}
+
+//nolint: whitespace
+func (mongodService *MongodService) GetBackupCursorExtended(backupID *primitive.Binary,
+	lastTS primitive.Timestamp) (*mongo.Cursor, error) {
+
+	return mongodService.MongoClient.Database(adminDB).Aggregate(mongodService.Context, mongo.Pipeline{
+		{{
+			Key: "$backupCursorExtend", Value: bson.D{
+				{Key: "backupId", Value: backupID},
+				{Key: "timestamp", Value: lastTS},
+			},
+		}},
+	})
+}
+
+func (mongodService *MongodService) FixSystemDataAfterRestore(LastWriteTS primitive.Timestamp) error {
+	ctx := mongodService.Context
+	localDatabase := mongodService.MongoClient.Database("local")
+
+	err := replaceData(ctx, localDatabase.Collection("replset.election"), true, nil)
+	if err != nil {
+		return err
+	}
+
+	err = replaceData(ctx, localDatabase.Collection("replset.minvalid"), true, bson.M{
+		"_id": primitive.NewObjectID(),
+		"t":   -1,
+		"ts":  primitive.Timestamp{T: 0, I: 1},
+	})
+	if err != nil {
+		return err
+	}
+
+	tracelog.DebugLogger.Printf("oplogTruncateAfterPoint: %v", LastWriteTS)
+	err = replaceData(ctx, localDatabase.Collection("replset.oplogTruncateAfterPoint"), true,
+		bson.M{
+			"_id":                     "oplogTruncateAfterPoint",
+			"oplogTruncateAfterPoint": LastWriteTS,
+		})
+	if err != nil {
+		return err
+	}
+
+	err = replaceData(ctx, localDatabase.Collection("system.replset"), false, nil)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func replaceData(ctx context.Context, collection *mongo.Collection, drop bool, insertData bson.M) error {
+	findCursor, err := collection.Find(ctx, bson.D{})
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("unable to find all from %v", collection.Name()))
```

Comment:
`errors.Wrap` is deprecated, we use `fmt.Errorf` in the new code.

Reply:
Why? errors. Wrap will take stacktrace for good troubleshooting

Reply:
> Why? errors. Wrap will take stacktrace for good troubleshooting

`errors` package is no longer maintained. If you want to keep stacktraces, use `errors.Errorf`.